spec/views/shared/activities/activity_spec.rb

```diff
@@ -200,11 +208,48 @@
         end
       end
     end
+
+    context "when the activity is an ISPF programme" do
+      let(:activity) { build(:programme_activity, :ispf_funded) }
+
+      context "and it doesn't have a linked activity" do
+        before { render }
+
+        it "shows a link to add a linked activity" do
+          expect(body.find(".linked_activity .govuk-summary-list__actions a")).to have_content(t("default.link.add"))
+        end
+      end
+
+      context "and it has a linked activity" do
+        let(:linked_prog) { create(:programme_activity, :ispf_funded) }
+        let(:activity) { build(:programme_activity, :ispf_funded, linked_activity: linked_prog) }
```

Comment:
I think this could also just be (one line?):

```ruby
let(:linked_prog) { build(:programme_activity, :ispf_funded, linked_activity: activity) }
```

Likewise in the next example, the two `let`s could probably be one, making use of the outer `context`'s `activity`

Reply:
I think this is the only outstanding thing, but fine to ignore! I guess it should actually use `create`:

```ruby
let(:linked_prog) { create(:programme_activity, :ispf_funded, linked_activity: activity) }
```